docs/backstage.md

```diff
@@ -0,0 +1,124 @@
+# Backstage
+
+## Overview
+
+Backstage is an open platform for building developer portals. It was created by Spotify to streamline their development processes and has since been open-sourced. Backstage allows you to manage all your infrastructure, services, and tools in one place, providing a unified developer experience.
+
+## Features
+
+- **Service Catalog**: Organize and manage all your services.
+- **Software Templates**: Standardize and automate the creation of new projects.
+- **TechDocs**: Centralize your documentation.
+- **Plugins**: Extend Backstage with a wide range of plugins.
+
+## Context in This Project
+
+In this project, Backstage is used to provide a unified developer portal that integrates various tools and services. It helps in managing the infrastructure and services more efficiently. The project leverages Backstage to:
+
+- **Centralize Documentation**: Using TechDocs to keep all documentation in one place.
+- **Manage Services**: Using the Service Catalog to organize and manage microservices.
+- **Automate Workflows**: Using Software Templates to standardize project creation.
+
+## Optional - Building Backstage Image
+This repo uses a hosted backstage image with entra auth enabled, automatically onboarding users into your backstage user list. It also has an example software catalog template to demo creating the resources required for argo to create and bootstrap a cluster named by the user. If you want to test Backstage please continue to getting started. 
+
+If you want to make changes to this image such as adding a different domain or new software catalogs you will need to make your changes, build your own image and change the deployment manifest to reference the image you have created. The source code for Backstage is found in the root Backstage folder. To build the image follow the steps below:
+
+1. **Fork & Clone the Repository**:
+    - First, fork the repository to your own GitHub account by clicking the "Fork" button on the repository page.
+    - Then, clone your forked repository:
+    ```sh
+    git clone https://github.com/<your_fork>/aks-platform-engineering.git
+    cd aks-platform-engineering
+
+2. **Install Dependencies**: Ensure all dependencies are installed by running:
+
+    ```sh
+    yarn install
+    ```
+
+3. **Build the Project**: Run the build script defined in your `package.json`. Based on your previous commands, it looks like you need to build the backend:
+
+    ```sh
+    yarn build:backend --config ./app-config-local.yaml
+    ```
+
+4. **Optional - Run the Application Locally**: After building the project, you can run it locally. Ensure that all necessary environment variables are set:
+
+    ```sh
+    export BASE_URL=https://your-local-base-url.com
+    export POSTGRES_HOST=your-local-postgres-host
```

Comment:
Is there any easy local docker command someone can run to get this going?

Reply:
Possibly i'd have to give it some more thought.